ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/JoinNode.java

```diff
@@ -67,6 +72,20 @@
     INNER, LEFT, OUTER
   }
 
+  private static final ImmutableList<KeyProperty> KEY_PROPERTIES = ImmutableList.of(
+      new KeyProperty("format", kf -> kf.getFormatInfo().getFormat()),
+      new KeyProperty("properties", kf -> kf.getFormatInfo().getProperties()),
+      new KeyProperty("features", KeyFormat::getFeatures),
+      new KeyProperty("widowing", kf -> kf.getWindowInfo()
+          .map(WindowInfo::getType)
+          .map(t -> t == WindowType.SESSION
+              ? "session windowing" : "non session windowing")
+          .orElse("no windowing"))
+  // Note: we deliberately do NOT fail if one side is HOPPING and the other is TUMBLING, or fail
+  // on different window sizes. This is because we only fail if the _schema_ of the key is
+  // different. Neither of these differences make the schemas different.
```

Comment:
Why are the schemas for session and non-session windows different? Is this because non-session windows don't store window end times?

Reply:
Correct